cpu/cortexm_common/vectors_cortexm.c

```diff
@@ -119,8 +119,19 @@ void nmi_default(void)
 /* The hard fault handler requires some stack space as a working area for local
  * variables and printf function calls etc. If the stack pointer is located
  * closer than HARDFAULT_HANDLER_REQUIRED_STACK_SPACE from the lowest address of
- * RAM we will reset the stack pointer to the top of available RAM. */
-#define HARDFAULT_HANDLER_REQUIRED_STACK_SPACE (THREAD_EXTRA_STACKSIZE_PRINTF)
+ * RAM we will reset the stack pointer to the top of available RAM.
+ * Measured from trampoline entry to breakpoint:
+ *  - Cortex-M0+ 344 Byte
+ *  - Cortex-M4  344 Byte
+ */
+#define HARDFAULT_HANDLER_REQUIRED_STACK_SPACE          (344U)
+
+static inline int _stack_size_left(uint32_t required)
+{
+    uint32_t* sp;
+    asm volatile ("mov %[sp], sp" : [sp] "=r" (sp) : : );
+    return ((int)((uint32_t)sp - (uint32_t)&_sstack) - required);
```

Comment:
> it looks kind of messy with all the subtractions of unsigned values and casting to signed.

Indeed. But I guess all the casting is neccessary. Do you have a proposal on how to improve?